### PR TITLE
Introduce branch protection via settings.yml

### PR DESCRIPTION
**Pull request by svrnm**
@lelia

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -84,36 +84,35 @@ repository:
...
       # enforce_admins: true
       # Prevent merge commits from being pushed to matching branches
-      # required_linear_history: true
+      required_linear_history: true
       # Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.
-      # restrictions:
-        # apps: []
-        # users: []
-        # teams:
-        #  - cisco-open/homebrew-tap-admins
```
Comment (lelia) [inline .github/settings.yml:119]: Just double-checking that the removal of the team here was intentional?

Comment (svrnm) [reply .github/settings.yml:119]: There was a discussion between me and @detiber if this is needed. Reason I put this here was initially that I didn't want teams to self-approve their PRs but by doing that we would bottleneck them ... let me link the discussion

Comment (svrnm) [reply .github/settings.yml:119]: https://github.com/cisco-open/homebrew-tap/pull/33#discussion_r1324551736

Comment (lelia) [reply .github/settings.yml:119]: Ty for the context!

**Review by lelia — APPROVED** [review]

**Pull request merged**